internal/resources/azure/application_gateway.go

```diff
@@ -0,0 +1,192 @@
+package azure
+
+import (
+	"github.com/infracost/infracost/internal/resources"
+	"github.com/infracost/infracost/internal/schema"
+
+	"fmt"
+	"strings"
+
+	"github.com/infracost/infracost/internal/usage"
+	"github.com/shopspring/decimal"
+)
+
+type ApplicationGateway struct {
+	Address                string
+	Region                 string
+	SKUName                string
+	SKUCapacity            int64
+	MonthlyDataProcessedGB *float64 `infracost_usage:"monthly_data_processed_gb"`
+	MonthlyV2CapacityUnits *int64   `infracost_usage:"monthly_v2_capacity_units"`
+}
+
+var ApplicationGatewayUsageSchema = []*schema.UsageItem{{Key: "monthly_data_processed_gb", ValueType: schema.Float64, DefaultValue: 0}, {Key: "monthly_v2_capacity_units", ValueType: schema.Int64, DefaultValue: 0}}
+
+func (r *ApplicationGateway) PopulateUsage(u *schema.UsageData) {
+	resources.PopulateArgsWithUsage(r, u)
+}
+
+func (r *ApplicationGateway) BuildResource() *schema.Resource {
+	var monthlyDataProcessedGb, monthlyCapacityUnits *decimal.Decimal
+	var sku, tier string
+	costComponents := make([]*schema.CostComponent, 0)
+	tierLimits := []int{10240, 30720}
+
+	skuNameParts := strings.Split(r.SKUName, "_")
+	if len(skuNameParts[1]) != 0 {
```

Comment:
This could throw an index error, so we should check the length here and log a warning and return an unsupported resource if the SKU isn't the expected format.

Reply:
@sinabakh I still see this file migrated. Is this file just being kept, but not referenced in the `azure/registry`? If so we should probably add comments against those in the registry to say we've already done some of the migration so we don't get confused.